src/frontend/src/common/analytics/methods.ts

```diff
@@ -0,0 +1,184 @@
+ * app's initial load. We fire off first `identify` method (assuming opt-in)
+ * as part of the SegmentInitializer -> segmentInitScript interaction. By
+ * relying on the init script to handle first `identify`, we can avoid the
+ * race condition.
+ *   IMPORTANT: Because of ^^^ above, we must ensure that the `identify` call
+ *   in segmentInitScript matches the structure of how `identify` is used here
+ *   and that SegmentInitializer is pulling that same data to pass to script,
+ *   otherwise the structure of the `identify` payload will change depending
+ *   on where it came from and it will cause problems for our data analyst.
+ */
+let previouslySentUserInfo: AnalyticsUserInfo = {};
+function analyticsSendUserInfo_(userInfo: User): void {
+  const analyticsUserInfo = extractAnalyticsUserInfo(userInfo);
+  if (isEmpty(previouslySentUserInfo)) {
+    // Very first user info payload handled by Segment init script.
+    // If this is empty now, record info, but skip doing anything.
+    previouslySentUserInfo = analyticsUserInfo;
+  } else if (
+    window.analytics &&
+    window.isCzGenEpiAnalyticsEnabled &&
+    !isEqual(previouslySentUserInfo, analyticsUserInfo)
+  ) {
+    window.analytics.identify(analyticsUserInfo.anonUserId, {
+      group_id: analyticsUserInfo.groupId,
+      group_name: analyticsUserInfo.groupName,
+    });
+    previouslySentUserInfo = analyticsUserInfo;
+  }
+}
+// Sending user info (ignoring app load `identify` call) usually happens based
+// on `useUserInfo` calls. Since these tend to bunch up and get called one on
+// top of the other, we debounce so we don't spam analytics.
+export const analyticsSendUserInfo = debounce(analyticsSendUserInfo_, 500, {
+  trailing: true,
+});
+
+/**
+ * Record page change in Segment. Meant for use in Nextjs router.events.
+ *
+ * The Nextjs `router` allows you to subscribe to various events. This func is
+ * subscribed to the routeChangeComplete event, so whenever the route changes,
+ * we kick off an analtyics call and report new page to Segment (assuming that
+ * the user has opted in to analytics, if not, it's a no-op).
+ *
+ * While this should be a simple process as outlined above, we currently have
+ * an issue where the router fires off lots of duplicate "changes" when the
+ * app is loading. See this ticket for more details:
+ *   FIXME: https://app.shortcut.com/genepi/story/204580
+ * To avoid spamming a bunch of duplicate page change events, we keep a var
+ * of the previouslyRecordedRoute: if a new route "change" matches the "change"
+ * we just saw, we ignore it and do nothing. This prevents firing duplicate
+ * `page` calls until we can fix whatever is causing the weird router stuff.
+ */
+let previouslyRecordedRoute = "";
+export function analyticsRecordRouteChange(pageUrl: string): void {
+  if (
+    window.analytics &&
+    window.isCzGenEpiAnalyticsEnabled &&
+    previouslyRecordedRoute !== pageUrl
+  ) {
+    window.analytics.page();
```

Comment:
do you need to pass `pageUrl` to `window.analytics.page`?

Reply:
Nah, it just pulls it automatically